frontend/src/app/testResults/viewResults/ResultsTable.tsx

```diff
@@ -214,45 +202,117 @@ const generateResultRows = (
 
 interface ResultsTableListProps {
   results: Array<Result>;
-  setPrintModalId: Dispatch<SetStateAction<Maybe<string> | undefined>>;
-  setMarkCorrectionId: Dispatch<SetStateAction<Maybe<string> | undefined>>;
-  setDetailsModalId: Dispatch<SetStateAction<Maybe<string> | undefined>>;
-  setTextModalId: Dispatch<SetStateAction<Maybe<string> | undefined>>;
-  setEmailModalTestResultId: Dispatch<
-    SetStateAction<Maybe<string> | undefined>
-  >;
   hasMultiplexResults: boolean;
   hasFacility: boolean;
 }
 
+type ActionInformation = {
+  modalType: "NONE" | "PRINT" | "CORRECTION" | "TEXT" | "DETAILS" | "EMAIL";
+  testResult: Result | undefined;
+};
+
 const ResultsTable = ({
   results,
-  setPrintModalId,
-  setMarkCorrectionId,
-  setDetailsModalId,
-  setTextModalId,
-  setEmailModalTestResultId,
   hasMultiplexResults,
   hasFacility,
 }: ResultsTableListProps) => {
+  /**
+   * Modals and selected test result handlers
+   */
+
+  const [actionSelected, setActionSelected] = useState<ActionInformation>({
+    modalType: "NONE",
+    testResult: undefined,
+  });
+
+  const dismissModal = () => {
+    setActionSelected(
+      (prevState): ActionInformation => ({ ...prevState, modalType: "NONE" })
+    );
+  };
+
+  const setFocusOnActionMenu = (actionName: string) => {
+    const buttonSelector = `#action_${actionSelected?.testResult?.id}`;
+    const printButtonSelector = `#${actionName}_${actionSelected?.testResult?.id}`;
+    waitForElement(buttonSelector).then((actionButton) => {
+      (actionButton as HTMLElement)?.click();
+      waitForElement(printButtonSelector).then((printButton) => {
+        (printButton as HTMLElement)?.focus();
+        const event = new MouseEvent("mouseover", {
+          view: window,
+          bubbles: true,
+          cancelable: true,
+        });
+        printButton?.dispatchEvent(event);
+      });
+    });
+  };
+
+  /**
+   * HTML
+   */
+
   return (
-    <table className="usa-table usa-table--borderless width-full">
-      <thead className="sr-element__sr-only">
-        {generateTableHeaders(hasFacility)}
-      </thead>
-      <tbody data-testid={"filtered-results"}>
-        {generateResultRows(
-          results,
-          setPrintModalId,
-          setMarkCorrectionId,
-          setDetailsModalId,
-          setTextModalId,
-          setEmailModalTestResultId,
-          hasMultiplexResults,
-          hasFacility
-        )}
-      </tbody>
-    </table>
+    <div title="filtered-result">
+      <table className="usa-table usa-table--borderless width-full">
+        <thead className="sr-element__sr-only">
+          {generateTableHeaders(hasFacility)}
+        </thead>
+        <tbody data-testid={"filtered-results"}>
+          {generateResultRows(
+            results,
+            setActionSelected,
+            hasMultiplexResults,
```

Comment:
I don't think we are actually needing `hasMultiplexResults` anymore if you agree then I can create a separate ticket to clean this up!

Reply:
I removed the `hasMultiplexResults` altogether so no need of a ticket. Thanks!